rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp

```diff
@@ -114,6 +115,72 @@ class TimeControllerClockImpl
     snapshot(ros_now());
   }
 
+  /**
+   * \brief Adjust internal clock to the specified timestamp.
+   * \details It will change the current internally maintained offset so that next published time
+   * is different.
+   * \note Will trigger any registered JumpHandler callbacks.
+   * \param ros_time Time point in ROS playback timeline.
+   */
+  void jump(rcutils_time_point_value_t ros_time)
+  {
+    rcl_duration_t time_jump_delta;
+    {
+      std::lock_guard<std::mutex> lock(state_mutex);
+      time_jump_delta.nanoseconds = ros_time - steady_to_ros(now_fn());
+    }
+
+    rcl_time_jump_t time_jump{};
+    time_jump.clock_change = RCL_ROS_TIME_NO_CHANGE;
+    time_jump.delta = time_jump_delta;
+
+    process_callbacks_before_jump(time_jump);
+    {
+      std::lock_guard<std::mutex> lock(state_mutex);
+      snapshot(ros_time);
+    }
+    process_callbacks_after_jump(time_jump);
+    cv.notify_all();
+  }
+
+  /**
+   * \brief Add callbacks to be called when a time jump exceeds a threshold.
+   * \param handler Shared pointer to the JumpHandler object returned from JumpHandler::create(..)
+   * \throws std::invalid argument if jump threshold has invalid value.
+   */
+  void add_jump_callbacks(PlayerClock::JumpHandler::SharedPtr handler)
+  {
+    if (handler->notice_threshold.min_forward.nanoseconds < 0) {
+      throw std::invalid_argument("forward jump threshold must be positive or zero");
+    }
+    if (handler->notice_threshold.min_backward.nanoseconds > 0) {
+      throw std::invalid_argument("backward jump threshold must be negative or zero");
+    }
+
+    std::lock_guard<std::mutex> lock(callback_list_mutex_);
+    for (auto const & registered_handler : callback_list_) {
+      if (*registered_handler == *handler) {
+        return;  // Already have this callback in the list.
+      }
+    }
+    callback_list_.push_back(handler);
+  }
+
+  /**
+   * \brief remove jump callbacks from processing list.
+   * \param handler Shared pointer to the JumpHandler object returned from JumpHandler::create(..)
+   */
+  void remove_jump_callbacks(PlayerClock::JumpHandler::SharedPtr handler)
+  {
+    std::lock_guard<std::mutex> lock(callback_list_mutex_);
+    for (auto it = callback_list_.begin(); it != callback_list_.end(); ++it) {
```

Comment:
The function name and comment make it sound like this will remove multiple jump callbacks from the processing list. However, the code only removes one. Should the comment/function name change to make it more clear that we're removing one callback?

Reply:
A `JumpHandler` contains multiple callback functions. Maybe `remove_jump_handler` would be more clear - but the language is consistent